src/Command/ExplainAdminCommand.php

```diff
@@ -102,20 +102,8 @@ public function execute(InputInterface $input, OutputInterface $output)
             ));
         }
 
-        $metadata = false;
-
-        if ($this->getContainer()->has('validator.validator_factory')) {
-            $factory = $this->getContainer()->get('validator.validator_factory');
-
-            if (method_exists($factory, 'getMetadataFor')) {
-                $metadata = $factory->getMetadataFor($admin->getClass());
-            }
-        }
-
-        // NEXT_MAJOR: remove method check in next major release
-        if (!$metadata) {
```

Comment:
This was a mistake, it broke https://github.com/sonata-project/SonataAdminBundle/issues/5016 , `$metadata = $this->getContainer()->get('validator')->getMetadataFor($admin->getClass());` this one should been saved.